### DISCUSSION TITLE
Building ALVR server failing

### DISCUSSION THREAD

**Discussion by ZombiesFromPluto** [gfx-rs/wgpu#2147] [Q&A]
Hi. I don't know what I am doing, but I am trying to build ALVR for linux.
The command I run is  "cargo xtask build-server --release --bundle-ffmpeg"
and I get an error:
error: failed to get `wgpu` as a dependency of package `client v16_0_0-rc2 (root/ALVR/alvr/experiments/client)`
I have absolutely no clue what this means and would appreciate if anyone could help me out.
Thanks.

**Comment by kvark** [comment]
"Failure to get" sounds like an issue with dependency setup or cargo environment. E.g. is ALVR using vendored dependencies or a custom crates io repository? I have no idea what ALVR is. Please provide more info:
  1. is there a downstream issue?
  2. what is the exact log?
  3. what are the repro steps?

**Reply by ZombiesFromPluto** [reply]
ALVR is a server that you can connect to with a standalone virtual reality headset like the oculus quest and stream pc virtual reality games to the headset.
Sorry, I don't know what a downstream issue is or how to get the log. I am very new to linux but here is the alvr github: https://github.com/alvr-org/ALVR

The reproduction steps are (in a terminal):

run "sudo apt install curl"
run "curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs/ | sh"
when asked enter "1"
run "sudo apt install build-essential pkg-config libclang-dev libssl-dev libasound2-dev libgtk-3-dev libvulkan-dev libunwind-dev gcc-8 g++-8 yasm nasm curl libx264-dev libx265-dev libxcb-render0-dev libxcb-shape0-dev libxcb-xfixes0-dev libspeechd-dev libxkbcommon-dev libdrm-dev"
4: wait a long time for the installation to complete.
5: run "cd ALVR"
run "cargo xtask build-server --release --bundle-ffmpeg"

Thanks.

**Reply by kvark** [reply]
Thank you for the info!
The log I ask is basically what your terminal outputs on the failing command.
The "downstream" here is the project using wgpu - ALVR. So I suggest filing a bug here.

Note that they have [CI in place](https://github.com/alvr-org/ALVR/runs/4060925263?check_suite_focus=true) and everything builds fine there.

Could your error be an intermittent network error on your side?

**Reply by ZombiesFromPluto** [reply]
Alright. I am downloading and reinstalling debian from scratch and running everything again to see if it was a network error. If it isn't I will submit a bug I guess.

**Reply by ZombiesFromPluto** [reply]
![Screenshot_2021-11-03-15-59-20-14_84d3000e3f4017145260f7618db1d683](https://user-images.githubusercontent.com/33584830/140004498-0c5747ce-4fb4-41d0-8456-038318370f2a.jpg)
Here is the log. It still doesn't work after reinstalling everything. I will submit a bug to ALVR.
Thanks so much for helping.

**Reply by kvark** [reply]
This revision does exist, so I'm not sure why it fails to fetch it: https://github.com/gfx-rs/wgpu/commit/5f1b3e56bc00eb3f805ffa8ae8ecde4a1bf09949
